### ISSUE TITLE
[SITE][FEATURE] Block merges on PRs that have a do-not-merge label

### ISSUE THREAD

**Issue by willihay** [o3de/o3de.org#1385]
## Describe the feature you'd like available on the website

This is a feature request for the o3de.org GitHub repo. If possible, the repo should block any attempt to merge a PR that has any of the do-not-merge labels. This is to avoid a potential source of confusion and accidental merges by well-intentioned reviewers. Since adding this label to a PR is an act of communication of status, it should be followed by a matching action to change the status.

## Proposed implementation

If possible, add a rule for the repo that checks for any label that begins with "do-not-merge" as part of the determination of the "ability to merge" status.

## Available alternatives

unknown

## Supplements

none

**Comment by FiniteStateGit** [comment]
I agree, this is good policy.  The ready-for-merge tag seems like a good choice for PRs that are reviewed but on hold for some reason, as generally the second reviewer to approve should merge a PR immediately.  Am I correct in that assumption?  Are there instances where you hold off on merging a PR that has no publishing date constraints?

**Comment by willihay** [comment]
I think the ready-for-merge tag should always be optional, as per the common practice you mentioned - where the second reviewer can generally go ahead and merge a PR immediately upon approval. I suppose ready-for-merge could be used by an author on a PR that had the do-not-merge tag for a long time, to sort of call attention to the fact that the PR is now in a state where it can be readily merged at the earliest convenience.

The do-not-merge tag should be used in cases where a non-draft PR should not be merged even if it has two or more approvals. This could be a publishing date constraint, or it could be some other condition that the author or a maintainer has in mind. For example, I've used this tag when I want to make sure a PR doesn't get merged until it has undergone a review by specific people, such as subject matter experts for a tech review or an editor for an editorial review.
